PayloadPkg/OsLoader/OsLoader.c

```diff
@@ -387,7 +387,7 @@ SetupBootImage (
     if (LoaderPlatformInfo != NULL) {
       if (FeaturePcdGet (PcdMeasuredBootEnabled) && (LoaderPlatformInfo->LdrFeatures & FEATURE_MEASURED_BOOT)) {
         // Extend hash of the image into TPM.
-        TpmExtendPcrAndLogEvent (8, TPM_ALG_SHA256, LoadedImage->ImageHash,
+        TpmExtendPcrAndLogEvent (8, (TPMI_ALG_HASH) GetTpmHashAlgEnabled(), LoadedImage->ImageHash,

```

Comment:
Will the ImageHash be of the same type as the PCR bank enabled in TPM?

Reply:
It would be different in some cases. As OsLoader uses container support, I could add condition to check hash type.